app/models/shelter.rb

```diff
@@ -4,6 +4,9 @@ class Shelter < ApplicationRecord
   validates :city, presence: true
 
   has_many :pets, dependent: :destroy
+  has_many :applications, through: :pets
```

Comment:
Nice work setting up this additional association! Adding this makes the syntax for joining below so much simpler. Mooching off of child model's associations is always a great choice!